apps/api/src/app/notifications/usecases/get-activity-feed/get-activity-feed.usecase.ts

```diff
@@ -46,6 +65,72 @@ export class GetActivityFeed {
     };
   }
 
+  private async validateRetentionLimitForTier(organizationId: string, after?: string, before?: string) {
+    const organization = await this.organizationRepository.findById(organizationId);
+
+    if (!organization) {
+      throw new HttpException('Organization not found', HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    const maxRetentionMs = this.getMaxRetentionPeriodByOrganization(organization);
+
+    const earliestAllowedDate = new Date(Date.now() - maxRetentionMs);
+
+    // If no after date is provided, default to the earliest allowed date
+    const effectiveAfterDate = after ? new Date(after) : earliestAllowedDate;
+    const effectiveBeforeDate = before ? new Date(before) : new Date();
```

Comment:
it would be better to rather have smallest possible range like `24h` by default instead of `earliestAllowedDate`, but someone might be trying to find event by `subscriberId` alone etc and since we don't provide any information to the user which range was used by default, he would think the event doesn't exist.

This shouldn't break any existing user endpoint usage, because although we returned "everything" if date range was not used, we still deleted the notifications using the background job. so last N days were always returned anyway.

Reply:
Agree - I added the information to DTO description.